ddtrace/opentelemetry/span_test.go

```diff
@@ -132,6 +134,61 @@ func TestSpanSetName(t *testing.T) {
 	assert.Equal(strings.ToLower("NewName"), p[0]["name"])
 }
 
+func TestSpanLink(t *testing.T) {
+	assert := assert.New(t)
+	t.Setenv("DD_TRACE_DEBUG", "true")
+
+	_, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	_, payloads, cleanup := mockTracerProvider(t)
+	tr := otel.Tracer("")
+	defer cleanup()
+
+	_, sp := tr.Start(context.Background(), "first_span")
+	sp.End()
+	tracer.Flush()
+	_, err := waitForPayload(payloads)
+	assert.NoError(err)
+
+	_, decoratedSpan := tr.Start(context.Background(), "span_with_link",
+		oteltrace.WithLinks(oteltrace.Link{
+			SpanContext: sp.SpanContext(),
+			Attributes:  []attribute.KeyValue{attribute.String("yes", "no")},
+		}))
+
+	decoratedSpan.End()
+	tracer.Flush()
+	payload, err := waitForPayload(payloads)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	assert.NotNil(payload)
+	assert.Len(payload[0][0], 1)
+
+	var spanLinks []ddtrace.SpanLink
+	spanLinkBytes, _ := json.Marshal(payload[0][0]["span_links"])
```

Comment:
Here I use `json.Marshal(...)` to convert `span["span_links"]` from `type interface{}` to a byte array, This byte array is then converted to an array of `ddtrace.SpanLink` structs. Not sure if this is the best approach